todoTask/todoTask.js

```diff
@@ -1,85 +1,59 @@
-﻿// ------------------------global varibals--------------------
-let globalID = 0;
-// -------------------------functions---------------------------
-const getId = () => (globalID += 1);
+const getId = (() => {
+    let globalID = 0;
+    return () => (globalID += 1);
+})();
 
-const makeTodo = (title, complete) => {
-    return {
-        id: getId(),
-        title,
-        complete,
-    };
-};
+const createTodo = (title, complete = false) => ({
+    id: getId(),
+    title,
+    complete,
+});
 
-const makeTodos = (n) => {
-    let todos = [];
-    for (let i = 1; i <= n; i++) {
-        let title = `title${i}`;
-        let complete = false;
-        let todo = makeTodo(title, complete);
-        todos = addTodo(todos, todo);
-    }
-    return todos;
-};
+const createTodos = (count) =>
+    Array.from({ length: count }, (value, i) => createTodo(`title${i}`));
 
-const addTodo = (todos, newTodo) => todos.concat(newTodo);
+const addTodo = (todos, newTodo) => [...todos, newTodo];
 
 const removeTodo = (todos, todoId) =>
     todos.filter((todoItem) => todoItem.id !== todoId);
 
 const removeCompleted = (todos) =>
     todos.filter((todoItem) => !todoItem.complete);
 
-const changeTodoTitle = (todos, todoId, title) => {
-    return todos.map((todoItem) =>
+const changeTodoTitle = (todos, todoId, title) =>
+    todos.map((todoItem) =>
         todoItem.id === todoId ? { ...todoItem, title } : todoItem
     );
-};
 
-const changeTodoComplete = (todos, todoId, complete) => {
-    return todos.map((todoItem) =>
+const changeTodoComplete = (todos, todoId, complete) =>
+    todos.map((todoItem) =>
         todoItem.id === todoId ? { ...todoItem, complete } : todoItem
     );
-};
-
-const editedTodo = (todos, todoEdited) => {
-    if (todoEdited) {
-        return todos.map((todoItem) =>
-            todoItem.id === todoEdited.id
-                ? { ...todoItem, ...todoEdited }
-                : todoItem
-        );
-    } else {
-        throw new Error("Entering edited object not correct or null");
-    }
-};
-//--------async function for state------------------
-const setState = async (todos = [], loading = true, error = new Error()) => {
-    const state = {
```

Comment:
No need to store `loading` state in any storage. Because it makes sense only exactly in app memory.
What "loading state" can be in server? In server data is always loaded